ray-operator/controllers/ray/common/service.go

```diff
@@ -40,6 +40,42 @@ func BuildServiceForHeadPod(cluster rayiov1alpha1.RayCluster, labels map[string]
 		annotations = make(map[string]string)
 	}
 
+	defaultAppProtocol := DefaultServiceAppProtocol
+	ports := getServicePorts(cluster)
+	if cluster.Spec.HeadGroupSpec.HeadService != nil {
```

Comment:
We may need to consider using deep copy instead. See these links ([link1](https://alenkacz.medium.com/shallow-copy-of-a-go-struct-in-golang-is-never-a-good-idea-83be60106af8), [link2](https://echorand.me/posts/go-values-references-etc/)) for more details.

Reply:
(Fixed)